File: internal/services/like.service.ts
import { ILikeRepo } from "#internal/interfaces/like.repo.interface.js";
import { ILikeService } from "#internal/interfaces/like.service.interface.js";

export class LikeService implements ILikeService{
    constructor(private readonly likeRepo: ILikeRepo){}

    async like(videoId: string, userId: string): Promise<void>{
        await this.likeRepo.like(videoId, userId)
    }

    async unlike(videoId: string, userId: string): Promise<void>{
        await this.likeRepo.unlike(videoId, userId)
    }

    async getLikesCount(videoId: string): Promise<number | null>{
        return await this.likeRepo.getLikesCount(videoId)
    }
}

File: src/main/java/com/matreshka/feed_service/application/LikeService.java
package com.matreshka.feed_service.application;

import com.matreshka.feed_service.application.port.ILikeService;
import com.matreshka.feed_service.delivery.http.dto.LikeRequestDTO;
import com.matreshka.feed_service.internal.repo.IVideoRepo;
import com.matreshka.feed_service.internal.repo.port.IVideoCacheRepo;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
@Slf4j
public class LikeService implements ILikeService {

    private final IVideoCacheRepo videoCacheRepo;
    private final IVideoRepo videoRepo;

    @Override
    public void like(LikeRequestDTO likeRequestDTO) {
        videoCacheRepo.like(likeRequestDTO.videoId());
    }

    @Override
    public void unlike(LikeRequestDTO likeRequestDTO) {
        videoCacheRepo.unlike(likeRequestDTO.videoId());
    }

    @Override
    public long getLikesCount(String videoId) {
        long dbLikes = videoRepo.getLikes(UUID.fromString(videoId));
        long cacheDelta = videoCacheRepo.getDelta(videoId);
        return dbLikes + cacheDelta;
    }
}
